java/Anton-Kozyk/src/com/zoolatech/lecture2/tasks/_2/Task2.java

```diff
@@ -0,0 +1,189 @@
+package com.zoolatech.lecture2.tasks._2;
+
+import java.util.Scanner;
+
+/**
+ * Define a class that represents a robot, which moves in a room of a width W and a height
+ * of H. The class should provide methods to move a robot forward by either 1 or N tiles
+ * and turn it either left or right by 90 degrees from the current direction. The class should
+ * also provide methods to get the robot's current location and direction.
+ * The north-west corner of a room is represented by coordinates (0,0); north-east - (W-1,
+ * 0); south-east - (W-1, H-1); south-west - (0, H-1). By default, a robot should be placed
+ * in the north-west corner facing south direction. However, it should be also possible to
+ * give any coordinates and direction as the initial robot location. Robot must not be able
+ * to get outside of a room - in such case the robot should move as much as possible to the
+ * room border, print a warning message and stop. All next moves in the same direction should
+ * be prevented.
+ */
+
+/**
+ When you first run the program, you will be asked to enter the dimensions of the
+ room (width and height) - they cannot be negative or zero. Next you will be able to enter the initial
+ coordinates of the robot, to do this, enter "y" and then specify the desired values,
+ keep in mind that these values can only be within the previously defined room
+ dimensions. To continue with the default values of the coordinates of the robot,
+ simply type any character and press Enter. Now the robot will follow your instructions
+ from the code. He can go forward a certain number of steps, or turn left and right. Also
+ note that left or right is relative to the direction of the robot. That is, if the robot
+ is standing in the direction of the South, turning to the left, it will turn in the
+ direction of the East. If the number of steps entered exceeds the size of the room, the
+ robot will stop at its border. printPosition() method displays the compass, current
+ position, direction, as well as a map with a picture of a robot on it (it can be displayed
+ in 4 ways, depending on the direction - [ > ], [ < ], [/\ ] , [\/ ]).
+ */
+
+public class Task2 {
+    public static void main(String[] args) {
+        Robot robot = new Robot();
+        robot.printPosition();
+        robot.moveForward(10);
+        robot.turnLeft();
+        robot.moveForward(4);
+        robot.turnLeft();
+        robot.moveForward(3);
+        robot.printPosition();
+        robot.turnLeft();
+        robot.printPosition();
+        robot.moveForward(4);
+        robot.printPosition();
+    }
+}
+
+class Robot {
+    //Start position coordinates
+    int x = 0;
+    int y = 0;
+    char direction = 'S';
+
+    int W; //width of area
+    int H; //height of area
+
+    { //Initialization block
```

Comment:
All input data should be accepted in the main method, and passed to the robot's constructor.